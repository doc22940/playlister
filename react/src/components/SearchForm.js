import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import DatePicker from 'react-datepicker';
import moment from 'moment';

import * as actions from '../modules/weather-actions';

import 'react-datepicker/dist/react-datepicker.css';


class SearchForm extends Component {
  static propTypes = {
    date: PropTypes.number.isRequired,
    fetchWeather: PropTypes.func.isRequired,
    location: PropTypes.string.isRequired,
    updateLocationInput: PropTypes.func.isRequired,
    updateDateInput: PropTypes.func.isRequired
  }

  constructor(props) {
    super(props);

    this.fetchWeather = this.fetchWeather.bind(this);
    this.updateDate = this.updateDate.bind(this);
    this.updateLocation = this.updateLocation.bind(this);
  }

  updateDate(date) {
    // assume a moment object
    this.props.updateDateInput(date.valueOf());
  }

  updateLocation(event) {
    this.props.updateLocationInput(event.target.value);
  }

  fetchWeather() {
    this.props.fetchWeather(this.props.location, this.props.date);
  }

  render() {
    const { location, date } = this.props;

    return (
      <form className='form-inline' action='#'>
        <input
          className='form-control mr-2 mb-2 mb-md-0'
          type='text'
          placeholder='Noblesville, IN'
          value={location}
          onChange={this.updateLocation}
        />
        <span
          id='date-picker-wrapper'
          className='mr-2 mb-2 mb-md-0'
          style={{ display: 'inline-block' }}
        >
          <DatePicker
            className='form-control'
            selected={moment(date)}
            onChange={this.updateDate}
          />
        </span>

        <button
          className='btn btn-outline-success mb-2 mb-md-0'
          type='button'
          onClick={this.fetchWeather}
        >
          Get Weather
        </button>
      </form>
    );
  }
}

export default connect(
  state => ({
    location: state.weather.locationInput,
    date: state.weather.dateInput
  }),
  dispatch => bindActionCreators({
    fetchWeather: actions.fetchWeather,
    updateLocationInput: actions.updateLocationInput,
    updateDateInput: actions.updateDateInput
  }, dispatch)
)(SearchForm);

