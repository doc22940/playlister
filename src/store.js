import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import createHistory from 'history/createBrowserHistory';
import createSagaMiddleware from 'redux-saga';

import rootReducer from './modules';
import sagas from './sagas';

export const history = createHistory();

const sagaMiddleware = createSagaMiddleware();
const initialState = {};
const enhancers = [];
const middleware = [routerMiddleware(history), sagaMiddleware];

if (process.env.NODE_ENV === 'development') {
  const { devToolsExtension } = window;

  if (typeof devToolsExtension === 'function') {
    enhancers.push(devToolsExtension());
  }
}

const composedEnhancers = compose(applyMiddleware(...middleware), ...enhancers);

const store = createStore(rootReducer, initialState, composedEnhancers);

sagaMiddleware.run(sagas);

export default store;

