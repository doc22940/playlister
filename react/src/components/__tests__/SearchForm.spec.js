import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux'

import SearchForm from '../SearchForm';
import store from '../../store';

test('Header snapshot', () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <SearchForm />
      </MemoryRouter>
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


