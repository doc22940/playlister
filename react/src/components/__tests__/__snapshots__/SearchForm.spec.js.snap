// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<form
  action="#"
  className="form-inline"
>
  <input
    className="form-control mr-2 mb-2 mb-md-0"
    onChange={[Function]}
    placeholder="Noblesville, IN"
    type="text"
    value="Indianapolis, IN"
  />
  <span
    className="mr-2 mb-2 mb-md-0"
    id="date-picker-wrapper"
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value="01/29/2018"
          />
        </div>
      </div>
    </div>
  </span>
  <button
    className="btn btn-outline-success mb-2 mb-md-0"
    onClick={[Function]}
    type="button"
  >
    Get Weather
  </button>
</form>
`;
